nextflow_schema.json

```
@@ -0,0 +1,470 @@
+                    "fa_icon": "fas fa-envelope",
+                    "help_text": "Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits. If set in your user config file (`~/.nextflow/config`) then you don't need to specify this on the command line for every run.",
+                    "pattern": "^([a-zA-Z0-9_\\-\\.]+)@([a-zA-Z0-9_\\-\\.]+)\\.([a-zA-Z]{2,5})$"
+                },
+                "bigwig": {
+                    "type": "boolean",
+                    "description": "Specifies if TSS-bigwigs should be generated, additionally to the TSS-bed files",
+                    "fa_icon": "fas fa-sign-out-alt"
+                }
+            }
+        },
+        "alignment_options": {
+            "title": "Alignment options",
+            "type": "object",
+            "description": "Adjust parameters and filtering criteria for read alignments.",
+            "default": "",
+            "fa_icon": "fas fa-map-signs",
+            "properties": {
+                "aligner": {
+                    "type": "string",
+                    "default": "star",
+                    "fa_icon": "fas fa-map-marked-alt",
+                    "description": "Alignment tool to be used",
+                    "enum": [
+                        "star",
+                        "bowtie1"
+                    ]
+                },
+                "min_aln_length": {
+                    "type": "integer",
+                    "default": "15",
+                    "fa_icon": "fas fa-ruler",
+                    "description": "Minimum number of aligned basepairs of a read to be kept"
+                }
+            }
+        },
+        "reference_genome_options": {
+            "title": "Reference genome options",
+            "type": "object",
+            "fa_icon": "fas fa-dna",
+            "description": "Options for the reference genome indices used to align reads.",
+            "properties": {
+                "genome": {
+                    "type": "string",
+                    "description": "Name of iGenomes reference.",
+                    "fa_icon": "fas fa-book",
+                    "help_text": "If using a reference genome configured in the pipeline using iGenomes, use this parameter to give the ID for the reference. This is then used to build the full paths for all required reference genome files e.g. `--genome GRCh38`.\n\nSee the [nf-core website docs](https://nf-co.re/usage/reference_genomes) for more details."
+                },
+                "fasta": {
+                    "type": "string",
+                    "fa_icon": "fas fa-font",
+                    "description": "Path to FASTA genome file.",
+                    "help_text": "If you have no genome reference available, the pipeline can build one using a FASTA file. This requires additional time and resources, so it's better to use a pre-build index if possible."
```

Is this the right help message? The pipeline can't build a 'genome reference' for you, that _is_ the FASTA. Maybe should be

```suggestion
                    "help_text": "If you have no genome reference index available, the pipeline can build one using a FASTA file. This requires additional time and resources, so it's better to use a pre-built index if possible."
```

But then that doesn't make sense for the `--fasta` parameter, which should just be to the reference FASTA. You should also describe what the reference FASTA should be of (e.g. a RNA database or a DNA reference genome of a particular species <- again I don't know this method so maybe I'm speaking out of ignorance).

---

Good point, I'll switch to RNA-seq help text for this parameter, because it is doing the same and we have a bit more consistency across nf-core docs.